protocol-units/da/m1/light-node/src/v1/sequencer.rs

```diff
@@ -97,39 +94,46 @@ impl LightNodeV1 {
 		}
 	}
 
-	async fn submit_blocks(&self, blocks: &Vec<Block>) -> Result<(), anyhow::Error> {
-		let uid = LOGGING_UID.load(Ordering::SeqCst);
-		info!(
-			target: "movement_timing",
-			batch_size = blocks.len(),
-			uid = uid,
-			"inner_submitting_block_batch"
-		);
-		let mut block_blobs = Vec::new();
+	async fn submit_blocks(&self, blocks: &Vec<block::WrappedBlock>) -> Result<(), anyhow::Error> {
 		for block in blocks {
-			let block_blob = self.pass_through.create_new_celestia_blob(
-				bcs::to_bytes(block)
-					.map_err(|e| anyhow::anyhow!("Failed to serialize block: {}", e))?,
-			)?;
-			block_blobs.push(block_blob);
-		}
-		for block in blocks {
-			info!(target: "movement_timing", block_id = %block.id(), "submitting_block");
+			info!(target: "movement_timing", block_id = %block.block.id(), "inner_submitting_block");
 		}
+		// get references to celestia blobs in the wrapped blocks
+		let block_blobs = blocks
+			.iter()
+			.map(|wrapped_block| &wrapped_block.blob)
+			.cloned() // hopefully, the compiler optimizes this out
```

Comment:
My hunch is that the compiler can't optimize this, it needs the original `Vec` for the logging iteration below.

Maybe this is the case, converse to what I previously suggested, for collecting the IDs in a temporary `Vec` with a non-desctructive pass, and then moving the blob fields into their own `Vec` with `into_iter`. `itertools` may have an adapter for this.

Reply:
A plain old loop with `push`ing two vectors would be even simpler.

Reply:
Let me take a look at something here...

Reply:
Yeah, my initial belief is that there wasn't really a way around this because we the client ultimately relies on taking a slice of owned `Blobs`. But, under the hood, the client must be serializing those blobs, which for most libs can be done with a reference. So, I was hoping the compiler could pick up on this. But, that doesn't seem correct. 

The id logging is not the issue I was contending with here; that's always pretty easy to move out. It's actually the heuristic and retries which require I provide a new distribution upon failure. The only ways I see to do this--because the third-party client takes and consumes the `Blobs`--is to clone or instead pass down a reference and ultimately clone. The latter results in less cloning because--as in this implementation--I don't have to clone the whole `WrappedBlock` only the `Blob`. So, I opted for that. 

I also tried playing around with the `Deref` trait on the `WrappedBlock`. But, I didn't get anything I liked.

Reply:
Ah I see, this is already with heuristics and blocks need to remain past the `submit_blocks` call in the failure case. On the other hand, the Celestia client takes a slice of blobs and we can't change that. In this case, fair to copy. Unless we can split blobs from IDs and pass them as parallel collections, though this will be ugly.